libs/util/nestjs/kafka/src/kafka.pubsub.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, OnModuleInit } from "@nestjs/common";
+import { Kafka, logLevel } from "kafkajs";
+import { KafkaPubSub } from "graphql-kafkajs-subscriptions";
+import { KafkaEnvironmentVariables } from "@amplication/util/kafka";
+import { randomUUID } from "crypto";
+
+const KAFKA_PUBSUB_TOPIC = "pubsub-topic";
+const KAFKA_PUBSUB_GROUP_ID_PREFIX = "pubsub-group-id";
+
+@Injectable()
+export class KafkaPubSubService implements OnModuleInit {
+  private pubsub: KafkaPubSub;
+
+  async onModuleInit() {
+    await this.initializePubSub();
+  }
+
+  private async initializePubSub(): Promise<void> {
+    const kafkaEnv = new KafkaEnvironmentVariables("");
+
+    const kafka = new Kafka({
+      logLevel: logLevel.ERROR,
+      clientId: kafkaEnv.getClientId() + `-${randomUUID()}`,
+      brokers: kafkaEnv.getBrokers(),
+      connectionTimeout: 25000,
+      retry: {
+        retries: 3,
+        maxRetryTime: 3000,
+      },
```

Comment:
kafka configurations should match the `createNestjsKafkaConfig` to allow a single common setup and potentially customised that on top if needed.